iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalGrammar.java

```diff
@@ -128,74 +133,66 @@ public static LexerlessGrammarBuilder createGrammarBuilder() {
 
   private static void punctuators(LexerlessGrammarBuilder b) {
     for (Punctuator p : Punctuator.values()) {
-      b.rule(p).is(SPACING, p.getValue()).skip();
+      b.rule(p).is(p.getValue()).skip();
     }
   }
 
   private static void lexical(LexerlessGrammarBuilder b) {
-
-    b.rule(WHITESPACE_OR_ESCAPED_LINE_BREAK).is(
-      b.skippedTrivia(b.regexp("(?:[" + LexicalConstant.WHITESPACE + "]|" + DockerLexicalConstant.LINE_BREAK + ")*+"))
-    );
-
-    b.rule(WHITESPACE_OR_LINE_BREAK).is(
-      b.skippedTrivia(b.regexp("[" + LexicalConstant.LINE_TERMINATOR + LexicalConstant.WHITESPACE + "]*+"))
-    );
-
+    b.rule(WHITESPACE).is(b.regexp("["+LexicalConstant.WHITESPACE+"]++"));
+    b.rule(WHITESPACE_OR_LINE_BREAK).is(b.skippedTrivia(b.regexp("["+LexicalConstant.WHITESPACE+LexicalConstant.LINE_TERMINATOR+"]++")));
+    b.rule(EOL).is(b.regexp("(?:"+DockerLexicalConstant.EOL+"|$)"));
     b.rule(SPACING).is(
-      WHITESPACE_OR_ESCAPED_LINE_BREAK,
-      b.zeroOrMore(
-        b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)), b.regexp(DockerLexicalConstant.EOL),
-        WHITESPACE_OR_ESCAPED_LINE_BREAK)
-    ).skip();
-
-    b.rule(INSTRUCTION_PREFIX).is(
-      WHITESPACE_OR_LINE_BREAK,
-      b.zeroOrMore(
-        b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)),
-        WHITESPACE_OR_LINE_BREAK)
+      b.oneOrMore(
+        b.firstOf(
+          b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)),
+          WHITESPACE_OR_LINE_BREAK
+        )
+      )
```

Comment:
```suggestion
      WHITESPACE_OR_LINE_BREAK
      b.zeroOrMore(
          b.commentTrivia(b.regexp(DockerLexicalConstant.COMMENT)),
          WHITESPACE_OR_LINE_BREAK
      )
```
IMO, this should look like this.